README.md

```diff
@@ -1,55 +1,187 @@
+The configuration file is read from `stdin` channel and also contains the name of the existing 
+utility that needs to be used. Therefore, the generic mock executable has enough information 
+about the utility and the particular test. The command would be similar to:
+
+```shell script
+./mockexec --cull-time 202007210000 -o output.dat input1.dat input2.dat input3.dat < config.textproto
+```
+
+### Input files
+
+The content of input files sometimes can be very large and comparing it directly can be
+very ineffective. To avoid this, we calculate the hash of the input content and check that it matches
+the hash of the expected input. We use the sha256 hash function.
+
+### Output file
+
+When the utility produces an output, the tool gives two options to indicate its content.
+Only one option can be chosen at the same time. You can either set the `out_path` field (used for larger content) 
+or you can set the `out_content` field (used for smaller content). 
+
+### Configuration file
+
+To generate the configuration files, we use the `BuildMockConfig` function, which takes the name of the binary
+and the arguments which are of certain types and builds the `MockCall` protocol buffer.
+
+The configuration file for each utility can be written manually by following a series of syntax rules. 
+To make this procedure easier, we have implemented the `genrconfig` tool. It uses the `BuildMockConfig` function 
+to produce the following configuration file and prints it to `stdout` channel.
+One example of use would be:
+
+```shell script
+./genrconfig example "strarg:-t" "strarg:SOME STRING" "strarg:-o" "outpath:tmp/test-output" "infile:tmp/test-input" > config.textproto
+```
+
+## Structure
 
-## Example
+### `/cmd`
 
-As an example of how to use this tool, we are going to mock the `tar` utility.
+Main functions for the project - `mockexec.go`, `genrconfig.go` and `bldmock.go`.
 
-[compress](https://github.com/googleinterns/executable-mocks/examples/tar/compress.sh) is a program that uses the `tar` utility. The program echoes some lines and stores some files in a new file.
+### `/configs`
 
-The tests of `compress` will be slow because they run `compress`, that executes the `tar` utility. However, the `tar` utility is already tested, so the `compress` tests would run it without need. To make them faster, we are going to mock the `tar` utility for the `compress` tests.
+The configuration files for example functions `util1` and `util2`.
+
+### `/examples`
+
+Simple example mocks of `util1` and `util2` along with the shell scripts for testing.
+A descriptive example of how to use the generic mock utility when mocking `tar` utility.
+
+### `/protos`
+
+The `.proto` file with all the message types. 
+
+### `/test`
+
+Tests for `mockexec.go`, `genrconfig.go` and `bldmock.go` along with the configuration files for them.
+
+### `/tools`
+
+Right now this directory only has a function to calculate the hash of an input file.
+
+## Getting Started
```

Comment:
I would move the Getting Started before the Design.

What you could do is take a small snippet of the description and make it into a section:
  ## Overview
it would be just one paragraph long
Then you can have the
 ## Getting Started
Then the rest you keep the order.

I think it makes sense to quickly show example usage and instructions, people often like to jump to that and try it out. So that's why I would try to keep a very short description first and then have examples, and only then the detailed design.

Reply:
Okay!